patches/server/0966-Call-BlockGrowEvent-for-missing-blocks.patch

```diff
@@ -0,0 +1,176 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Lulu13022002 <41980282+Lulu13022002@users.noreply.github.com>
+Date: Fri, 9 Jun 2023 13:04:42 +0200
+Subject: [PATCH] Call BlockGrowEvent for missing blocks
+
+Fixes some cancellability issues for sugar cane, cactus and melon/pumkin stems
+Include the source block when available
+
+diff --git a/src/main/java/net/minecraft/world/entity/animal/Bee.java b/src/main/java/net/minecraft/world/entity/animal/Bee.java
+index 55026e1731e41b4e3e4c6a8fef5d96a32051a556..98d1104fa42512264d72ddc3c9801a44013091ab 100644
+--- a/src/main/java/net/minecraft/world/entity/animal/Bee.java
++++ b/src/main/java/net/minecraft/world/entity/animal/Bee.java
+@@ -1247,8 +1247,8 @@ public class Bee extends Animal implements NeutralMob, FlyingAnimal {
+                         }
+ 
+                         if (iblockdata1 != null && CraftEventFactory.callEntityChangeBlockEvent(Bee.this, blockposition, iblockdata1)) { // CraftBukkit
++                            if (!org.bukkit.craftbukkit.event.CraftEventFactory.handleBlockGrowEvent(Bee.this.level(), blockposition, iblockdata1, 3)) continue; // Paper
+                             Bee.this.level().levelEvent(2005, blockposition, 0);
+-                            Bee.this.level().setBlockAndUpdate(blockposition, iblockdata1);
+                             Bee.this.incrementNumCropsGrownSincePollination();
+                         }
```

Comment:
Having the event here misses 2 instances of bee-caused growth. Cave vines don't go into this conditional block since `iblockdata1` will still be null.

Reply:
Yeah, I think its worth calling for berries. It is growing, but yeah, you can put it in the performBonemeal method.